zcash_client_sqlite/src/wallet.rs

```diff
@@ -1964,6 +1972,90 @@ pub(crate) fn store_transaction_to_be_sent<P: consensus::Parameters>(
         }
     }
 
+    // Add the transaction to the set to be queried for transaction status. This is only necessary
+    // at present for fully-transparent transactions, because any transaction with a shielded
+    // component will be detected via ordinary chain scanning and/or nullifier checking.
+    if !detectable_via_scanning {
+        queue_tx_retrieval(
+            wdb.conn.0,
+            std::iter::once(sent_tx.tx().txid()),
+            TxQueryType::Status,
+            None,
+        )?;
+    }
+
+    Ok(())
+}
+
+pub(crate) fn set_transaction_status(
+    conn: &rusqlite::Transaction,
+    txid: TxId,
+    status: TransactionStatus,
+) -> Result<(), SqliteClientError> {
+    match status {
+        TransactionStatus::TxidNotRecognized | TransactionStatus::NotInMainChain => {
+            // If the transaction is now expired, remove it from the retrieval queue.
+            if let Some(chain_tip) = chain_tip_height(conn)? {
+                let mut stmt = conn.prepare_cached(
+                    "SELECT expiry_height FROM transactions
+                     WHERE txid = :txid",
+                )?;
+
+                let expiry_height = stmt
+                    .query_row(named_params![":txid": txid.as_ref()], |row| {
+                        row.get::<_, Option<u32>>(0)
+                            .map(|opt| opt.map(BlockHeight::from))
+                    })
+                    .optional()?
+                    .flatten();
+
+                // If the transaction's expiry height + the confirmation depth is less than the
+                // chain tip height, we know it to be un-mineable and we can remove the reference
+                // to it from the transaction retrieval queue.
+                if expiry_height
+                    .iter()
+                    .any(|expiry| *expiry + VERIFY_LOOKAHEAD < chain_tip)
+                {
+                    conn.execute(
+                        "DELETE FROM tx_retrieval_queue WHERE txid = :txid",
+                        named_params![
+                            ":txid": txid.as_ref()
+                        ],
+                    )?;
+                }
```

Comment:
I think this could probably be expressed more simply as a single SQL statement, i.e. something like
```sql
DELETE FROM tx_retrieval_queue q
JOIN transactions t
ON q.txid = t.txid
WHERE t.txid = :txid AND t.expiry_height < :chain_tip_minus_lookahead
```

Reply:
Ah, yeah, this can be simplified now that it's no longer setting the `last_observed_unmined_height` field that had been added to the `transactions` table in a previous version of this PR but which has been removed as it ultimately was not needed.